frontend/src/lib/keybind.ts

```diff
@@ -0,0 +1,246 @@
+/**
+ * Enum to represent the key strings provided by {@link KeyboardEvent}
+ */
+export enum Keys {
+  J = 'j',
+  D = 'd',
+  K = 'k',
+  H = 'h',
+  V = 'v',
+  T = 't',
+  M = 'm',
+  L = 'l',
+  O = 'o',
+  S = 's',
+  SINGLE_QUOTE = "'",
+  FORWARD_SLASH = '/',
+  CTRL = 'Control',
+  ENTER = 'Enter',
+  ARROW_UP = 'ArrowUp',
+  ARROW_DOWN = 'ArrowDown',
+}
+
+// The type used to declare the key downs that will trigger the callback
+export interface KeybindCallback {
+  key: Keys;
+  callback: (e: KeyboardEvent) => void;
+  shift?: boolean;
+  ctrl?: boolean;
+  alt?: boolean;
+}
+
+/**
+ * Used to create and interface with key bindings
+ */
+export class Keybind {
+  // All of the keybindings to their callbacks
+  private bindings: KeybindCallback[];
+  private bindingEntries: Keys[];
+
+  /**
+   * Constructor
+   *
+   * @param bindings An array of {@link KeybindCallback}s to be evaluated and utilized with {@link Keybind:onKeyDown}.
+   */
+  constructor(bindings: KeybindCallback[]) {
+    this.bindings = bindings;
+    this.bindingEntries = this.bindings.map((x) => x.key);
+  }
+
+  /**
+   * Attempt to convert a string to {@link Keys} enum
+   *
+   * @param str - The string to attempt to convert
+   * @returns undefined if unable to convert, otherwise {@link Keys}
+   */
+  private keyEnumFromString(str: string): Keys | undefined {
+    for (const [_, key] of Object.entries(Keys)) {
+      if (key == str) return key;
+    }
+
+    return undefined;
+  }
```

Comment:
When `Key` is a type, this can be written as
```ts
private keyFromString(str: string): Key | undefined {
    return Keys.find((key) => key === str);
  }
  ```